xamarin/Xamarin.Forms
# [iOS,Android] Set Position and CurrentItem when ItemsSource items change

### Description of Change ###

When we add or remove items from the ItemsSource, we need to tell the new Position/CurrentItem on the Carousel. This change will listen to changes on Observable Collections and get the new position, when we set the Position we update the CurrrentItem.

Also resets the ItemDecoration on android when adding or removing items in the last position, since we add an extra spacing to those items so they can be centered. (#9401)

### Issues Resolved ### 

- fixes #9432 
- fixes #9322
- fixes  #9401 

### API Changes ###

Changed:
 -` public void SetCurrentItem(object item, int position = -1)`

### Platforms Affected ### 

- iOS
- Android

### Behavioral/Visual Changes ###


Should update CurrentItem

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###

Go to CarouselXamlGallery, swipe to the second item , click on the button to Remove make sure the Position and the CurrentItem change.

To test 9432 go to IndicatorCodeGallery , swipe to the last item, click remove, check that the Indicators update and the last one is still selected.

To test 9401 , visit the CarouselView / CarouselView (Indicators Forms), go to the last item, click add item,  make sure the added item doesn't show a extra spacing on the left of the last item.

### PR Checklist ###
